apidef/oas/operation.go

```diff
@@ -163,6 +166,133 @@ func (s *OAS) fillPathsAndOperations(ep apidef.ExtendedPathsSet) {
 	s.fillDoNotTrackEndpoint(ep.DoNotTrackEndpoints)
 	s.fillRequestSizeLimit(ep.SizeLimit)
 	s.fillRateLimitEndpoints(ep.RateLimit)
+	s.fillMockResponsePaths(s.Paths, ep)
+}
+
+// fillMockResponsePaths converts classic API mock responses to OAS format.
+// This method only handles direct mock response conversions, as other middleware
+// configurations (like allow lists, block lists, etc.) are converted to classic
+// API mock responses in an earlier step of the process.
+//
+// For each mock response, it:
+// 1. Creates an OAS operation with a unique ID (method + path + status code)
+// 2. Sets up the mock response with content type detection and example values
+// 3. Configures the operation to ignore authentication for this endpoint
+//
+// The content type is determined by:
+// - Checking the Content-Type header if present
+// - Attempting to parse the body as JSON
+// - Defaulting to text/plain if neither above applies
+func (s *OAS) fillMockResponsePaths(paths openapi3.Paths, ep apidef.ExtendedPathsSet) {
+	var methodOperationMap = map[string]func(*openapi3.PathItem, *openapi3.Operation){
+		http.MethodGet:     func(p *openapi3.PathItem, op *openapi3.Operation) { p.Get = op },
+		http.MethodPost:    func(p *openapi3.PathItem, op *openapi3.Operation) { p.Post = op },
+		http.MethodPut:     func(p *openapi3.PathItem, op *openapi3.Operation) { p.Put = op },
+		http.MethodPatch:   func(p *openapi3.PathItem, op *openapi3.Operation) { p.Patch = op },
+		http.MethodDelete:  func(p *openapi3.PathItem, op *openapi3.Operation) { p.Delete = op },
+		http.MethodHead:    func(p *openapi3.PathItem, op *openapi3.Operation) { p.Head = op },
+		http.MethodOptions: func(p *openapi3.PathItem, op *openapi3.Operation) { p.Options = op },
+	}
+
+	for _, mock := range ep.MockResponse {
+		pathItem := paths[mock.Path]
+		if pathItem == nil {
+			pathItem = &openapi3.PathItem{}
+			paths[mock.Path] = pathItem
+		}
+
+		operation := &openapi3.Operation{
+			Responses: openapi3.NewResponses(),
+		}
+
+		operation.OperationID = genMockResponseOperationID(mock)
+
+		// Response description is required by the OAS spec, but we don't have it in Tyk classic.
+		// So we're using a dummy value to satisfy the spec.
+		dummyDescription := "CHANGE ME"
+
+		response := &openapi3.Response{
+			Headers:     make(openapi3.Headers),
+			Description: &dummyDescription,
+		}
+
+		contentType := "text/plain"
+
+		for name, value := range mock.Headers {
+			if http.CanonicalHeaderKey(name) == "Content-Type" {
+				contentType = value
+				break
+			}
+		}
+
+		// We attempt to guess the content type by checking if the body is a valid JSON.
+		if mock.Body != "" {
+			var body json.RawMessage
+			if err := json.Unmarshal([]byte(mock.Body), &body); err == nil {
+				contentType = "application/json"
+			}
```

Comment:
this would also satisfy a simple string - [ref](https://go.dev/play/p/8BUaeNbdwIC)

How about we use a map to unmarshal? - [ref](https://go.dev/play/p/OrwjK7Rip7-)

Reply:
You're right, thanks @jeffy-mathew.
A quick question, do you think we should also unmarshal to []interface{}, to cover arrays?